api/types.proto

```diff
@@ -465,12 +465,31 @@ enum TaskState {
 	// Remove this message when the states are deemed perfect.
 }
 
+// State of Networks attached to a container
+message ContainerNetworkStatus {
```

Comment:
I noticed that this would be used in flat list, where the same values of `network` and `interface` may be repeated to provide multiple addresses.

If one of the networks is showing an error, there would be a single entry for it with an error instead of an IP.

This seems different from the way networks are represented internally in the executor. There's a set of networks at the top level, and each network can have zero or more IPs underneath it. Would it make sense to replicate that hierarchy here? My first thought is it would be a little less confusing than the flattened list here with the `oneof` choosing between an IP or an error condition, but I'm not really familiar with CNI, so it's very possible I'm missing something.